src/scripts/main.js

```diff
@@ -1,3 +1,43 @@
+/* eslint-disable no-console */
+/* eslint-disable no-shadow */
 'use strict';
 
-// write code here
+function convertSalaryToNumber(salary) {
+  return Number(salary.replace(/[^0-9.-]+/g, ''));
+}
+
+function sortList(list) {
+  const listItems = Array.from(list.children);
+
+  listItems.sort((a, b) => {
+    const salaryA = convertSalaryToNumber(a.dataset.salary);
+    const salaryB = convertSalaryToNumber(b.dataset.salary);
+
+    return salaryB - salaryA;
+  });
+  list.innerHTML = '';
+  listItems.forEach(item => list.appendChild(item));
+}
+
+function getEmployees(list) {
+  const employees = [];
+
+  Array.from(list.children).forEach(item => {
+    employees.push({
+      name: item.textContent.trim(),
+      position: item.dataset.position,
+      salary: item.dataset.salary,
+      age: item.dataset.age,
+    });
+  });
```

Comment:
Consider using a functional programming approach like 'map' instead of 'forEach' to create the 'employees' array. This would make your code more concise and easier to understand.